tests/keras/layers/cudnn_recurrent_test.py

```diff
@@ -463,5 +474,64 @@ def test_cudnnrnn_bidirectional():
     model.fit(x, y, epochs=1, batch_size=1)
 
 
+@keras_test
+@pytest.mark.parametrize('layer_class,layer_args', [
+    (keras.layers.GRU, {'units': 2, 'input_shape': [3, 5]}),
+    (keras.layers.GRU, {'units': 2, 'input_shape': [3, 5], 'reset_after': True}),
+    (keras.layers.LSTM, {'units': 2, 'input_shape': [3, 5]}),
```

Comment:
It would be good to keep the non-cudnn stuff back in `test_topology.py` and call the test method with `skipif` and different args for cudnn layers, however I'm not sure if such cross-test import would work since `tests/` do not have module structure. If we only want to allow running plain GRU/LSTM variants on Travis, we can just split it into two functions within the same file, possibly just the original `test_topology.py`, even though some classes would be cudnn-related.

Reply:
Fixed.

For some strange reason`test_preprocess_weights_for_loading_rnn_should_be_idempotent()` with GRU `{'reset_after': True}` fails when called without test parallelization `py.test -n 0 tests/keras/engine/test_topology.py::test_preprocess_weights_for_loading_rnn_should_be_idempotent` on `reset_after` being propagated to RNN kwargs. Without `-n 0` it behaves as expected, ie. `reset_after` is consumed as a named arg and not in kwargs. WTF? :)